examples/http_server_shutdown.cc

```diff
@@ -0,0 +1,48 @@
+#include "pistache/endpoint.h"
+#include <signal.h>
+
+using namespace Pistache;
+
+class HelloHandler : public Http::Handler {
+public:
+
+    HTTP_PROTOTYPE(HelloHandler)
+
+    void onRequest(const Http::Request& request, Http::ResponseWriter response) override{
+        UNUSED(request);
+        response.send(Pistache::Http::Code::Ok, "Hello World\n");
+    }
+};
+
+int main() {
+    sigset_t signals;
+    if (sigemptyset(&signals) != 0
+            || sigaddset(&signals, SIGTERM) != 0
+            || sigaddset(&signals, SIGINT) != 0
```

Comment:
I suggest also trapping SIGHUP for this example program.

Reply:
Well, it's make a sense.